### ISSUE TITLE
Datatable: empty `TH` tags should be `TD`

### ISSUE THREAD

**Issue by Eehol** [primefaces/primefaces#4725]
Hello,

For accessibility purpose (WCAG2.1 rule 1.3.1), the HTML tables should have a not empty `th` tags to make the column headers visible for screen reader.

PrimeFaces generates nicely the `th` but there is an exception. If a column contains only technical stuff (s.t. icons), its headerText could be empty. In that case, the following code is generated by `<p:column>` with PrimeFaces 7.0 :
```
<th id="myForm:table:j_idt102" class="ui-state-default" role="columnheader" scope="col">
        <span class="ui-column-title"></span>
</th>
```

As said previously, the `th` can't be empty. So when the `headerText/f:facet header` is blank, is it possible for PrimeFaces to generate a `td` instead of the `th` (only for that column) ?

Please note that this solution is also the one suggested by WAVE : 

> If the table cell is a header, provide text within the cell that describes the column or row. If the cell is not a header or must remain empty (such as the top-left cell in a data table), make the cell a `td `rather than a `th`.

An alternative solution exists : the developer can provide an accessibility-only title to the column such that `<span class="hidden">Action icons column</span>`. Personally I think that the choice between both solutions should be let to the developer because the second one is not always adapted.

**Comment by tandraschko** [comment]
Please provide a small reproducer

**Comment by Eehol** [comment]
For this sample : 
```
<html xmlns="http://www.w3.org/1999/xhtml" xmlns:h="http://java.sun.com/jsf/html" xmlns:p="http://primefaces.org/ui">
    <h:head />
    <h:body>
        <h:form>
        	<p:dataTable value="#{sample.cars}" var="car">
        		<p:column headerText="Car"><h:outputText value="#{car}" /></p:column>
        		<p:column><p:button value="View"/></p:column>
        	</p:dataTable>
        </h:form>
    </h:body>
</html>

```
```
@Named
@RequestScoped
public class Sample {
    public List<String> getCars() {
        List<String> list = new ArrayList<>();
        list.add("Citroën");list.add("BMW"); list.add("Toyota");
        return list; }}

```
PrimeFaces generates this HTML :
```
<table ... >
	<thead ... >
		<tr role="row">
			<th ... >
				<span class="ui-column-title">Car</span>
			</th>
			<th ... >
				<span class="ui-column-title"/>
			</th>
		</tr>
	</thead>
	<tbody ... > ...  </tbody>
</table>
```

But the second `<th>` is empty because it's a column filled with action buttons, so it should be a `<td>` to avoid a screen reader confusion :
```
<table ... >
	<thead ... >
		<tr role="row">
			<th ... >
				<span class="ui-column-title">Car</span>
			</th>
			<td ... >
				<span class="ui-column-title"/>
			</td>
		</tr>
	</thead>
	<tbody ... > ...  </tbody>
</table>
```

**Comment by melloware** [comment]
Just checking are you sure its valid to have TD inside THEAD?

**Comment by melloware** [comment]
Instead of changing to TD would just removing role="columnheader" scope="col" from the TH for these columns achieve the same thing?

**Comment by Eehol** [comment]
> Just checking are you sure its valid to have TD inside THEAD?

I've tried on the [W3CValidator](https://validator.w3.org/#validate_by_input) and it's valid (with the content of the attached file test.txt).

> Instead of changing to TD would just removing role="columnheader" scope="col" from the TH for these columns achieve the same thing?

We made a POC with an empty `th` without those attributes and it remains non compliant :-/ 
[test.txt](https://github.com/primefaces/primefaces/files/3062721/test.txt)

**Comment by melloware** [comment]
OK thanks for researching.

**Comment by melloware** [comment]
Looking at the code this is a much more difficult change than hoped.

**Comment by Rawi01** [comment]
I had a similar problem last week. According to our accessibility test we should use `td` instead of `th` for our "Toggle filters" button that toggles out filter row. This row should also use `td` for all cells. 

I created a custom renderer that supports rendering a single column and a full row as `td`. I can provide a pull request as soon as I have some time to refactor my stuff.

**Comment by melloware** [comment]
We already had a column attrbribute for `ariaHeaderText` to displaying screen reader text on columns that do not have a header.

For example...

```xml
<p:column style="width:2rem" ariaHeaderText="Row Toggler">
   <p:rowToggler/>
</p:column>
```

Generates this HTML with `hidden-accessible` so we don't have empty `TH` tags.

```xml
<th id="j_id_7j:j_id_7k:j_id_7m" class="ui-state-default" aria-label="Row Toggler" scope="col" style="width:2rem">
   <span class="ui-column-title ui-helper-hidden-accessible">Row Toggler</span>
</th>
```
